ui/plugin-system/src/components/PluginEditor/plugin-editor-api.ts

```diff
@@ -32,11 +32,14 @@ export interface PluginEditorProps extends Omit<BoxProps, OmittedMuiProps> {
 }
 
 type PreviousSpecState = Record<string, Record<string, UnknownSpec>>;
+type HideQueryEditorState = Record<string, boolean>;
 
 /**
  * Props needed by the usePluginEditor hook.
  */
-export type UsePluginEditorProps = Pick<PluginEditorProps, 'pluginType' | 'value' | 'onChange'>;
+export type UsePluginEditorProps = Pick<PluginEditorProps, 'pluginType' | 'value' | 'onChange'> & {
+  onHideQueryEditorChange?: (isHidden: boolean) => void;
```

Comment:
This change is needed so the `PanelSpecEditor` knows when to show/hide the query editor